Inline RTCSessionDescriptionInit description

bc-lee opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

Add missing explanation of RTCSessionDescriptionInit and RTCLocalSessionDescriptionInit dictionaries. Also correct several links referencing those interfaces.
### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests
Fixes #33962

bc-lee committed:
fix: Add missing documentation for RTCSessionDescriptionInit, RTCLocalSessionDescriptionInit

Add missing documentation for RTCSessionDescriptionInit,
RTCLocalSessionDescriptionInit interfaces. Also correct several links
referencing those interfaces. Also add a clear explanation of the
relationship between RTCSessionDescription, RTCSessionDescriptionInit,
and RTCLocalSessionDescriptionInit interfaces.

Fixes #33962

bc-lee committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

skyclouds2001 reviewed (commented):
I think the two dictionaries are better not to placed in separate pages, as they both only contains two properties and both only used in one method

Josh-Cena reviewed (changes requested):
Hi! Unfortunately we don't document dictionaries on MDN, unless they are reused by a lot of methods (and even in such case we would usually just copy the property list around). Could you inline the description in the places where they are used instead?

bc-lee committed:
Apply Code Review

Combine RTCSessionDescriptionInit and RTCLocalSessionDescriptionInit
descriptions within the RTCSessionDescription page.
Additionally, update any related links.

bc-lee committed:
Update files/en-us/web/api/rtcsessiondescription/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Josh-Cena reviewed (commented):
We never mention the dictionary's name at all, because it's not a code entity. It should look like this:

```
  - : An object with the following properties:
    - `property1`
      - : What it does
```

And then for the errors, you would just say "the `sessionDescription` parameter is invalid".

bc-lee commented:
Do you mean I should update the PR to remove any references to the directory name `RTCSessionDescriptionInit`? It existed before I submitted my PR.

Josh-Cena commented:
Yes, they should be removed.

bc-lee commented:
I'll keep my PR open for now, but I'm not sure if I can make any more changes, especially considering that the distinction of dictionary names is required since they are part of the WebRTC API. I believe it is important to retain the dictionary name in the documentation. If someone wants to take over the PR, I'm happy to close it.

Josh-Cena commented:
Please see past changes where dictionaries were removed, for example https://github.com/mdn/content/pull/30080, https://github.com/mdn/content/pull/30079

hamishwillee commented:
@bc-lee FYI @Josh-Cena is correct - the fact that the spec mentions these dictionaries is irrelevant. Specs are written for browser developers, while MDN interprets the specification for webapp developers.

In most cases we've found that having separate dictionaries is unhelpful for web app developers - they have to jump around to find relevant information, and often the compatibility data for the dictionary depends on where it is used. Because of that, the best way to properly track compatibility is to expand the dictionary.

We're not dogmatic about it. For some APIs it does actually make sense to define the objects passed to APIs as dictionaries. Generally this is when you have a general API that can take numerous possible objects depending on the context in which it is used.
This is not the case for WebRTC.

We won't be merging this as is. I'm sure it will be useful though if modified to match MDN conventions.

bc-lee commented:
Two days ago, a pull request relevant to this one (#34051) was merged. I'd like to hear from @Josh-Cena and @hamishwillee on whether you believe the merged PR achieves the same level of consistency as requested in this pull request.

Josh-Cena commented:
@bc-lee, do you have time to come back to this?

bc-lee commented:
Yes, but not today or tomorrow (atm I'm busy with other things). I'll come back to this in a few days. 

https://github.com/mdn/content/pull/33955#issuecomment-2284985438

Josh-Cena commented on files/en-us/web/api/rtcsessiondescription/rtcsessiondescription/index.md:
```diff
@@ -45,7 +45,7 @@ This example uses the constructor to convert an SDP offer into an
 
 > **Note:** This is no longer necessary, however;
-> {{domxref("RTCPeerConnection.setLocalDescription()")}} and other methods which take
+> [`RTCSessionDescriptionInit`](/en-US/docs/Web/API/RTCSessionDescription#rtcsessiondescriptioninit) and other methods which take
```
```suggestion
> {{domxref("RTCPeerConnection.setLocalDescription()")}} and other methods which take
```

Josh-Cena committed:
Update files/en-us/web/api/rtcsessiondescription/rtcsessiondescription/index.md

Josh-Cena committed:
Update

Josh-Cena committed:
Merge branch 'main' into feature/rtcsessiondescriptioninit

Josh-Cena committed:
Update files/en-us/web/api/rtcpeerconnection/setlocaldescription/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Josh-Cena committed:
Fix

Josh-Cena committed:
Fix others

Josh-Cena reviewed (approved):
I've simply inlined the descriptions in each page and reordered some sections. I did not do other major changes. Thanks for the PR!
